### PR TITLE
🌱 ignore cluster condition updates when enqueueing events

### PR DESCRIPTION
**Pull request by apricote**
**What this PR does / why we need it**:

Ignore cluster condition updates when enqueueing events.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to #926 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squash commits
- [ ] include documentation
- [ ] add unit tests

### CODE REVIEW

File: controllers/hcloudmachine_controller.go [hunk 1]
```diff
@@ -294,3 +301,57 @@ func (r *HCloudMachineReconciler) HetznerClusterToHCloudMachines(_ context.Conte
...
+
+			oldCluster.ResourceVersion = ""
+			newCluster.ResourceVersion = ""
+
+			oldCluster.Status.Conditions = nil
+			newCluster.Status.Conditions = nil
+
+			if reflect.DeepEqual(oldCluster, newCluster) {
+				// Only insignificant fields changed, no need to reconcile
+				log.V(4).Info("Update to resource only changes insignificant fields, will not enqueue event")
```
Comment (janiskemper) [inline controllers/hcloudmachine_controller.go:344]: we don't use V(4) and higher anywhere. If you wanna include it, then use V(1).

**Review by janiskemper — APPROVED** [review]
added a small comment about the log level. Otherwise
LGTM

File: controllers/hcloudmachine_controller.go (hunk 1 above)
Comment (apricote) [reply controllers/hcloudmachine_controller.go:344]: :+1: Copied those levels from the "not-paused" predicate in the CAPI repo. Will update in the next push.

Comment (apricote) [reply controllers/hcloudmachine_controller.go:344]: I do not think these need to be in the default logging level, but its helpful if one can enable these through a flag. Whats the level I should use for that?

Comment (apricote) [reply controllers/hcloudmachine_controller.go]: IIUC
- `V(1)` equals the Zap debug logging, which is the default in caph
- `V(0)` would be Zap info logging

I set both the `V(1)` and rebased.

**Comment by apricote** [comment]
I will add some tests in a follow up PR.

**Comment by janiskemper** [comment]
thanks :)

**Pull request merged**